OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -281,7 +283,12 @@ public void ResolveOrder(Actor self, Order order)
 					}
 
 					var valued = unit.TraitInfoOrDefault<ValuedInfo>();
-					var cost = valued != null ? valued.Cost : 0;
+					var cost = 0;
+					if (variedCostManager != null && valued != null && valued.Varies)
```

Comment:
Varies shouldn't be able to be true if the variedCostManager doesn't exist. These null checks don't look quite right.

Reply:
It currently can, with this code, if VariedCostManager is null but Varies is true, it'll ignore the Varies tag and no cost variation will happen. Should i crash or write a debug.log entry/lint error instead in that case?

Reply:
That's an obvious case of modder error, so silently approving it doesn't sit well with me. Crash the game IMO.